src/test/java/org/opensearch/security/auditlog/integration/BasicAuditlogTest.java

```diff
@@ -551,6 +551,70 @@ public void testIndexPattern() throws Exception {
         Assert.assertTrue(res.getBody().contains("auditlog-20"));
     }
 
+    @Test
+    public void testDataStreamAsBackend() throws Exception {
```

Comment:
I think we are missing a testing datatream's `_rollover` action - can you add validation for this kind of case where audit entries are created before/during/after and we are sure they have arrived intact?